### PR TITLE
finishedCreateUserModel

### PR DESCRIPTION
**Pull request by sarross88**
## Description

_1) Created the user model for backend using mongoose 

## Type of Changes

2 -File creation/ Model creation

### CODE REVIEW

File: models/UserModel.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import mongoose from 'mongoose'
+
+const UserSchema = new mongoose.Schema({
+  name: String,
+  email: String,
+  password: String,
+  lastName: {
+    type: String,
+    default: 'lastName',
+  },
```
Comment (GinaCastromonte) [inline models/UserModel.js]: For best practice, add a required email and password when creating a user.  Also, I would remove the default to the last name.

**Review by up2code3 — COMMENTED** [review]
1) Do we need want the hash/salt password stuff with the userSchema.pre("save" async...  . 

2) for email, should we add the unique:true for the email, to prevent duplicates in the database. 

3)also for email  the match: Regex check for a valid email address? 

Or is this more of a starting point for the user model, I have no idea how much or little should go into each ticket. I am not 100% on how or what I should be looking for on these approvals. FYI, Either way is totally fine with me.

**Comment by vsrodriguez3** [comment]
> 1. Do we need want the hash/salt password stuff with the userSchema.pre("save" async...  .
> 2. for email, should we add the unique:true for the email, to prevent duplicates in the database.
> 
> 3)also for email the match: Regex check for a valid email address?
> 
> Or is this more of a starting point for the user model, I have no idea how much or little should go into each ticket. I am not 100% on how or what I should be looking for on these approvals. FYI, Either way is totally fine with me.

Hi Tristan, I just added all 3 things you mentioned above to the User model.

**Review by up2code3 — APPROVED** [review]
Looks Great! Everything worked as expected.

**Comment by GinaCastromonte** [comment]
@vsrodriguez3 , @up2code3 .  Great job!  There are two things I would consider updating this for best practice.  I would separate the routes with the controller function.  Define controllers in a separate controller file called "authController.js".  Also I would add middleware to check roles and permissions if the app has admin or specific user access.

**Comment by vsrodriguez3** [comment]
> @vsrodriguez3 , @up2code3 . Great job! There are two things I would consider updating this for best practice. I would separate the routes with the controller function. Define controllers in a separate controller file called "authController.js". Also I would add middleware to check roles and permissions if the app has admin or specific user access.

Hi Gina, thank you for your feedback. I have implemented the changes you suggested.

**Comment by sarross88** [comment]
I pulled the latest, and ran npm install so I have the packages, just pushed up those changes

**Review by GinaCastromonte — APPROVED** [review]
Looks good- Just fix the merge conflicts and you're good to go

**Pull request merged**